File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minm=prices[0]
        max1=0
        for i in range(len(prices)):
            if prices[i]<minm:
                minm=prices[i]
            elif prices[i]-minm>max1:
                max1=prices[i]-minm
            # for j in range(i+1,len(prices)):
            #     profit=prices[j]-prices[i]
            #     max1=max(profit,max1)
        return max1
            

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int lowest=Integer.MAX_VALUE;
        int high=Integer.MIN_VALUE;
        int j=0;
        int profit=0;
        for(int i=0;i<n-1;i++){
            if(prices[i+1]-prices[i]>0){
                lowest = Math.min(prices[i],lowest);
                high = prices[i+1];
                profit = Math.max(profit,high-lowest);
            }
        }
        return profit;
    }
}